master/docs/manual/cfg-reporters.rst

```diff
@@ -899,3 +899,84 @@ It uses private token auth, and the token owner is required to have at least rep
     :param string baseURL: the base url of the GitLab host, up to and optionally including the first `/` of the path. Do not include /api/
     :param string verbose: Be more verbose
 
+
+.. bb:reporter:: HipchatStatusPush
+
+HipchatStatusPush
+~~~~~~~~~~~~~~~~~
+
+.. @cindex HipchatStatusPush
+.. py:class:: buildbot.reporters.hipchat.HipchatStatusPush
+
+::
+
+    from buildbot.plugins import reporters
+
+    hs = reporters.HipchatStatusPush('private-token', endpoint='https://chat.yourcompany.com')
+    c['services'].append(hs)
+
+:class:`HipchatStatusPush` publishes a custom message using `Hipchat API v2 <https://www.hipchat.com/docs/apiv2>`_.
+The message is published to a user and/or room in Hipchat,
+
+It requires `txrequests`_ package to allow interaction with Hipchat API.
+
+It uses API token auth, and the token owner is required to have at least message/notification access to each destination.
+
+
+.. py:class:: HipchatStatusPush(auth_token, endpoint="https://api.hipchat.com",
+                                builder_room_map=None, builder_user_map=None,
+                                wantProperties=False, wantSteps=False, wantPreviousBuild=False, wantLogs=False)
+
+    :param string auth_token: Private API token with access to the "Send Message" and "Send Notification" scopes.
+    :param string endpoint: (optional) URL of your Hipchat server. Defaults to https://api.hipchat.com
+    :param dictionary builder_room_map: (optional) If specified, will forward events about a builder (based on name) to the corresponding room ID.
+    :param dictionary builder_user_map: (optional) If specified, will forward events about a builder (based on name) to the corresponding user ID.
+    :param boolean wantProperties: (optional) include 'properties' in the build dictionary
+    :param boolean wantSteps: (optional) include 'steps' in the build dictionary
+    :param boolean wantLogs: (optional) include 'logs' in the steps dictionaries.
+        This needs wantSteps=True.
+        This dumps the *full* content of logs.
+    :param boolean wantPreviousBuild: (optional) include 'prev_build' in the build dictionary
+
+
+.. note::
+
+   No message will be sent if the message is empty or there is no destination found.
+
+.. note::
+
+   If a builder name appears in both the room and user map, the same message will be sent to both destinations.
+
+
+Json object spec
+++++++++++++++++
+
+The default json object sent is as follows:
+
+.. code-block:: json
+
+    {
+        "message": "Buildbot started/finished build MyBuilderName (with result success) here: http://mybuildbot.com/#/builders/23",
+        "id_or_email": "12"
+    }
+
+
+These are the minimum required parameters to send a message to Hipchat. If you require more don't hesitate to subclass
+and modify one or more of the following functions:
+:py:func:`getRecipientList` :py:func:`getMessage`, :py:func:`getExtraParams`
```

Comment:
Please describe each of those APIs with complete signature, and what they are supposed to return, etc.